testsuite/cluster-test/src/tx_emitter.rs

```diff
@@ -67,12 +68,46 @@ struct StatsAccumulator {
     latency: AtomicU64,
 }
 
+pub struct AtomicHistogram {
+    buckets: Vec<AtomicU64>,
+}
+
+impl AtomicHistogram {
+    fn default() -> Self {
+        let mut buf = Vec::with_capacity(HISTOGRAM_CAPACITY);
+        for _i in 0..HISTOGRAM_CAPACITY {
+            buf.push(AtomicU64::new(0));
+        }
+        Self { buckets: buf }
+    }
+
+    fn to_u64_array(&self) -> Vec<u64> {
+        let mut buf = Vec::with_capacity(HISTOGRAM_CAPACITY);
+        for _i in 0..HISTOGRAM_CAPACITY {
+            buf.push(self.buckets[_i].load(Ordering::Relaxed));
+        }
+
+        buf
+    }
+}
+
+impl Clone for AtomicHistogram {
```

Comment:
I think we should be able to make implementation of this without need to have Clone for histogram